api/datadoghq/v2alpha1/datadogagent_types.go

```diff
@@ -170,6 +170,88 @@ type SingleStepInstrumentation struct {
 	// Injector configures the APM Injector.
 	// +optional
 	Injector *InjectorConfig `json:"injector,omitempty"`
+
+	// Targets is a list of targets to apply the auto instrumentation to. The first target that matches the pod will be
+	// used. If no target matches, the auto instrumentation will not be applied. Full config key:
+	// apm_config.instrumentation.targets
```

Comment:
we don't have a precedence of referencing the datadog.yaml config keys in this file; do you think it's needed?

(in general, a goal of the operator is to obscure some of the complicated config when setting up the agent)